sdk/lib/opentelemetry/sdk.rb

```diff
@@ -61,6 +61,11 @@ module SDK
     #       c.use_all
     #     end
     def configure
+      if ENV['OTEL_SDK_DISABLED'] == 'true'
```

Comment:
@robertlaurin is this the best place for us to check this? 

Does it make sense to check that in the instance method instead? https://github.com/open-telemetry/opentelemetry-ruby/blob/main/sdk/lib/opentelemetry/sdk/configurator.rb#L144

Do we want to support being able to change the envar and re-initialize in a running program?

Reply:
Here in our applications works like a charm in this place (using monkey patch).

The PR content was based on this comment: https://github.com/open-telemetry/opentelemetry-ruby/discussions/1605#discussioncomment-8568551

Reply:
Yes, that was my comment 😄 

I reached out to @robertlaurin here since he designed this component and wanted to solicit his input about where this check should happen. 

If a user is _not_ using the DSL, they can still create an instance of the `Configurator` and call the `configure` method bypassing the `envar`. 

They could also not use the configurator at all and assemble the tracer provider and resources themselves. 

I just want to make sure we are hooking this into the appropriate scope, so folks are not surprised that it does not work as anticipated in specific scenarios; or if we are OK with this minimal implementation.

Reply:
I had similar questions @kaylareopelle to the ones you brought up

Reply:
thank you! missed this! [(link to comment)](https://github.com/open-telemetry/opentelemetry-ruby/pull/1773#issuecomment-2521597058)